### ISSUE TITLE
[BUG]: Docker does not build neither does conda env file 

### ISSUE THREAD

**Issue by gaussiangit** [CosmiQ/solaris#461]
so many conflicts in env.yml which results in conda not creating
Can anyone update env.yml ?

Even docker gets stuck at solving env. When I add strict priority, it goes but end up with tons of conflicts

**Comment by IAINATTWATER** [comment]
same here, is anyone listening....? Thanks.

**Comment by remtav** [comment]
@IAINATTWATER  I reached out to development seed this morning trying to get in touch with @rbavery . We (Natural Resources Canada) benefit from solaris quite a bit, but if we can't collaborate with issues, PRs, etc. then we'll have to look elsewhere.

**Comment by remtav** [comment]
@gaussiangit meanwhile, did you check the last developments? https://github.com/CosmiQ/solaris/blob/0.5.0/ROADMAP.md
I guess @rbavery will soon drop the current main branch and replace it with 0.5.0.­..
EDIT: this conda environment is **not** working for me on Ubuntu 20.04: https://github.com/CosmiQ/solaris/blob/0.5.0/environment.yaml
I've create a PR to fix this: #462

**Comment by IAINATTWATER** [comment]
Hi @remtav Thank you - I had not read that either. What caught my eye was the geojson to coco work and is what drew me to solaris in the first place; will be helpful for instance segmentation work. I will try the different environment yaml.

**Comment by remtav** [comment]
@IAINATTWATER environment seems broken actually. Check #462

**Comment by rbavery** [comment]
hi @remtav and others! I recently took on maintainership of this project in order to fix this and other issues with solaris and make it simpler to install, since it wasn't maintained for some time before this. That work is largely addressed in this release branch but I haven't had the time recently to push this through to merge to main: https://github.com/CosmiQ/solaris/tree/0.5.0

I think instead of moving installs to the pip section of the env file like https://github.com/CosmiQ/solaris/pull/462, we should keep them all installed by conda to benefit from a single solver, like in https://github.com/CosmiQ/solaris/blob/0.5.0/environment.yaml

I'll have more time to push this through to completion in July. I think the final step is to get a new readthedocs account set up and hooked up to pypi so that when new releases are published, the docs get updated (we don't have access to the old account, see https://github.com/CosmiQ/solaris/issues/447). For now I suggest using the 0.5.0 branch, any issues you find there and report I can help address, when I have time.

**Comment by IAINATTWATER** [comment]
Hi @rbavery  - thank you for the update! I tried the release branch but needed to make one change on the env file. The packe could not resolve opencv-python>=4.1, so I removed the "-python" bt and it seemed to complete and installed cleanly. I tried a python shell and was able to import cv2 so believe all is well - please alert me if this is incorrect.

Thank you again for your help.

Cheers

**Comment by IAINATTWATER** [comment]
after further investigation I'm noticing some rearrangement of the API. I was particularly looking for the function that used to reside in the sol.data.coco namespace - geojson2coco function. Looks like the data ns does not now have a coco path and it looks like there are some functions related to coco in geo.py. Therefore is there a geojson2coco funtion now?

Thank you.

Cheers

**Comment by rbavery** [comment]
@IAINATTWATER  thanks for the update on the install! looks like that fixes it on the release branch. I'll submit a pr and merge to the release branch.  Can you open a new issue for the api question?

**Comment by rbavery** [comment]
a fix for this is merged here: https://github.com/CosmiQ/solaris/pull/464

I'll close this issue once 0.5.0 is released and merged to main

sidenote, as @IAINATTWATER noted in #465, 0.5.0 includes substantial changes to the solaris api. most of this is removing solaris functionality for model training and keeping it focused to preprocessing and postprocessing so that it is more maintainable and easy to plug in to actual ML frameworks (pytorch-lightning, keras, detectron2, fastai, etc.) We would appreciate feedback and discussion from solaris users on the roadmap here: https://github.com/CosmiQ/solaris/discussions/437  CC @remtav @srmsoumya
